src/Form.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react'
+import getFormData from './getFormData'
+
+export default class Form extends Component {
+  constructor(props) {
+    super(props)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  handleSubmit(event) {
+    event.preventDefault()
+    const newCard = getFormData(event.target)
+    const { type, addToCardList, editCardList, params } = this.props
+    const editIndex = parseInt(params.cardId, 10) - 1
+
+    switch (type) {
+      case 'new':
+        addToCardList(newCard)
+        break
+      case 'edit':
+        editCardList(newCard, editIndex)
+        break
+    }
+
+    window.location.hash = 'cards'
+  }
+
+  render() {
+    const { type, params, questionText, answerText, cardList } = this.props
+    const editIndex = parseInt(params.cardId, 10) - 1
+    const cardDetails = cardList[editIndex]
```

Comment:
Your form knows too much about how the rest of the app works. Try to stick to "Single Responsibility".